src/components/docs-page/index.tsx

```diff
@@ -1,27 +1,27 @@
 import { ReactElement } from 'react'
 import { MDXRemote } from 'next-mdx-remote'
 import defaultMdxComponents from '@hashicorp/platform-docs-mdx'
+import { useCurrentProduct } from 'contexts'
 import Placement from 'components/author-primitives/shared/placement-table'
 import NestedNode from 'components/author-primitives/waypoint/nested-node'
-import makeDocsAnchor from 'components/docs-anchor'
-
-/**
- * TODO: this will eventually be configurable by all products, not just waypoint
- */
-const additionalComponents = defaultMdxComponents({
-  product: 'waypoint',
-  additionalComponents: {
-    Placement,
-    NestedNode,
-    a: makeDocsAnchor('waypoint', ['docs', 'plugins', 'commands']),
-  },
-})
+import DocsAnchor from 'components/docs-anchor'
 
 /**
  * TODO: we may find that we want to allow additional components to be passed in
  * as a prop in the future
  */
 const DocsPage = (props): ReactElement => {
+  const currentProduct = useCurrentProduct()
+
+  const additionalComponents = defaultMdxComponents({
+    product: currentProduct.slug,
+    additionalComponents: {
+      Placement,
+      NestedNode,
+      a: DocsAnchor,
+    },
+  })
```

Comment:
This should be memoized based on `currentProduct.slug`, to ensure that the component references don't get re-created on re-render, potentially causing full remounts.

```suggestion
  const additionalComponents = useMemo(() => 
    defaultMdxComponents({
	    product: currentProduct.slug,
	    additionalComponents: {
	      Placement,
	      NestedNode,
	      a: DocsAnchor,
	    },
	  }), [currentProduct.slug])
```

Reply:
Made this change! ✅